src/main/java/org/spongepowered/common/gui/window/AbstractSpongeWindow.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.gui.window;
+
+import static com.google.common.base.Preconditions.checkState;
+
+import net.minecraft.entity.player.EntityPlayerMP;
+import net.minecraft.network.Packet;
+import net.minecraft.network.play.INetHandlerPlayServer;
+import net.minecraft.network.play.server.S2EPacketCloseWindow;
+import net.minecraft.util.BlockPos;
+import org.spongepowered.api.entity.living.player.Player;
+import org.spongepowered.api.gui.window.Window;
+import org.spongepowered.common.interfaces.IMixinEntityPlayerMP;
+
+import java.util.Optional;
+
+public abstract class AbstractSpongeWindow implements Window {
+
+    protected static final BlockPos VIRTUAL_POS = new BlockPos(-1, -1, -1);
```

Comment:
Depends. It's up to the subclasses for how they go about sending fake things. Where I've used it so far is where it's safe to do so. I am relying on the fact that `World#getBlockState` will simply return `Blocks.air.getDefaultState()` if the position is invalid, and that no exceptions are thrown where I use this position